### PR TITLE
Initial core code

### PR DESCRIPTION
**Pull request by pgjones**
Basic Spectra class and sphinx documentation.

### CODE REVIEW

**Comment by pgjones** [comment]
@ashleyrback the errors you see should now be fixed.

File: echidna/core/spectra.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        """ Initialise the spectra data container.
+
+        Args:
+          name (str): The name of this spectra
+
+        Attributes:
+          _data (:class:`numpy.ndarray`): The histogram of data
+          _name (str): The name of this spectra
+        """
+        self._data = numpy.ndarray(shape=(Spectra._energy_bins, Spectra._radial_bins, Spectra._time_bins), dtype=float)
```
Comment (ashleyrback) [inline echidna/core/spectra.py]: Just a suggestion, I think this line would read better if split over multiple lines e.g.

``` python
self._data = numpy.ndarray(shape=(Spectra._energy_bins,
                                  Spectra._radial_bins,
                                  Spectra._time_bins), 
                           dtype=float)
```

Also you could combine this line and the next using the `numpy.zeros` method.

File: echidna/core/spectra.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      _energy_high (float): Highest bin edge in MeV
+      _energy_bins (int): Number of energy bins
+      _radial_low (float): Lowest bin edge in mm
+      _radial_high (float): Highest bin edge in mm
+      _radial_bins (int): Number of raidal bins
+      _time_low (float): Lowest bin edge in years
+      _time_high (float): Highest bin edge in years
+      _time_bins (int): Number of time bins
+    """
+    _energy_low = 0.0 # MeV
```
Comment (ashleyrback) [inline echidna/core/spectra.py]: I like the units in the comment after the value

File: echidna/core/spectra.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+          raidus (float): Radial value to fill.
+          time (float): Time value to fill.
+          weight (float, optional): Defaults to 1.0, weight to fill the bin 
+          with.
+
+        Raises:
+          ValueError: If the energy, radius or time is beyond the bin limits.
+        """
+        if not Spectra._energy_low <= energy < Spectra._energy_high:
+            raise ValueError("Energy out of range")
```
Comment (ashleyrback) [inline echidna/core/spectra.py]: Would it be useful to add `spectra.fill:` to the front of the error messages, so we know where they are coming from?

Comment (pgjones) [inline echidna/core/spectra.py]: I think this is something python automatically does, i.e. it knows the trace back to the exception raise.

**Comment by ashleyrback** [comment]
Documentation looks fine now, merging!

**Pull request merged**